src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -152325,5 +152719,149 @@ public static object Test(object a)
         }
+
+        [Fact, WorkItem(52827, "https://github.com/dotnet/roslyn/issues/52827")]
+        public void MethodGroupConversion_NullableAttributes()
+        {
+            var source =
+@"#nullable enable
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+class Program
+{
+    static void Main()
+    {
+        Func<object> a1 = f1;
+        Func<object?> a2 = f2;
+        Func<object> a3 = f3;
+        Action<bool> a4 = f4;
+
+        [return: MaybeNull] object f1() => null;
+        [return: NotNull] object? f2() => null;
+        [DoesNotReturn] object f3() => throw null!;
+        void f4([DoesNotReturnIf(false)] bool b) { }
+    }
+}";
+            var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition, NotNullAttributeDefinition, DoesNotReturnAttributeDefinition, DoesNotReturnIfAttributeDefinition });
+            comp.VerifyDiagnostics(
+                // (9,27): warning CS8621: Nullability of reference types in return type of 'object f1()' doesn't match the target delegate 'Func<object>' (possibly because of nullability attributes).
+                //         Func<object> a1 = f1;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInReturnTypeOfTargetDelegate, "f1").WithArguments("object f1()", "System.Func<object>").WithLocation(9, 27),
+                // (15,43): warning CS8603: Possible null reference return.
+                //         [return: NotNull] object? f2() => null;
+                Diagnostic(ErrorCode.WRN_NullReferenceReturn, "null").WithLocation(15, 43)
+                );
+        }
+
+        [Fact, WorkItem(47167, "https://github.com/dotnet/roslyn/issues/47167")]
+        public void AnonymousFunctionConversion_Issue47167()
+        {
+            var source =
+@"#nullable enable
+D<int> d = (out int v) => v = 0;
+delegate void D<T>(out T? value);
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (2,12): warning CS8622: Nullability of reference types in type of parameter 'v' of 'lambda expression' doesn't match the target delegate 'D<int>' (possibly because of nullability attributes).
+                // D<int> d = (out int v) => v = 0;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInParameterTypeOfTargetDelegate, "(out int v) =>").WithArguments("v", "lambda expression", "D<int>").WithLocation(2, 12)
+                );
+        }
+
+        [Fact, WorkItem(45571, "https://github.com/dotnet/roslyn/issues/45571")]
+        public void AnonymousFunctionConversion_Issue45571()
+        {
+            var source =
+@"#nullable enable
+using System.Diagnostics.CodeAnalysis;
+
+delegate void MyDel<T>([AllowNull] T t);
+
+class C
+{
+    static void F<T>(T t, MyDel<T> f)
+    {
+        f(default);
+    }
+    static void Main()
+    {
+        F(new object(), maybeNull => maybeNull.ToString());
+    }
+}
+";
+            var comp = CreateCompilation(new[] { source, AllowNullAttributeDefinition });
```

Comment:
Should we also have a comment here that the missing warning will be handled in a follow-up?